c7n/output.py

```diff
@@ -323,6 +322,37 @@ def get_metadata(self):
         return res
 
 
+class JSONFormatter(logging.Formatter):
+    """
+    Convert Logs to JSON for more usable cloudwatch logs
+    https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html
+    """
+
+    def format(self, record):
+        json_log_object = {"type": "log",
+                           "logger": record.name,
+                           "log_time": datetime.fromtimestamp(record.created).isoformat(),
+                           "exc_info": record.exc_info,
```

Comment:
exc_info is not serializable so this will fail if there's an exception. you want to detect its presence and then augment the record with additional keys. also speaks to needing an additional test.

Reply:
good catch. I added in logic for that and better test cases for exceptions.